client/java/src/test/java/io/openlineage/client/transports/KinesisTransportTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import com.amazonaws.services.kinesis.producer.KinesisProducer;
+import com.amazonaws.services.kinesis.producer.UserRecord;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClient;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Properties;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+class KinesisTransportTest {
+  @Test
+  void clientEmitsKinesisTransport() throws IOException {
+    KinesisProducer producer = mock(KinesisProducer.class);
+    KinesisConfig config = new KinesisConfig();
+
+    Properties properties = new Properties();
+    properties.setProperty("MinConnections", "1");
+
+    config.setRegion("us-west-2");
+    config.setRoleArn("test-role");
```

Comment:
This should be optional - it's failing test https://app.circleci.com/pipelines/github/OpenLineage/OpenLineage/4949/workflows/081d660c-e5d6-432c-9eda-3d33087a8b23/jobs/62505?invite=true#step-103-68